dipy/core/tests/test_geometry.py

```diff
@@ -315,5 +317,24 @@ def test_dist_to_corner():
     assert_array_almost_equal(dist_to_corner(new_aff), pythagoras)
 
 
+def test_is_hemispherical():
+    # Smoke test the ValueError for non-3D vectors
+    assert_raises(ValueError, is_hemispherical, np.array(
+        [[1, 2, 3, 4], [5, 6, 7, 8]]
+    ))
+
+    # Test on hemispherical input
+    xyz = random_uniform_on_sphere(n=100, coords='xyz')
+    xyz = xyz[xyz[:, 2] > 0]
+    assert is_hemispherical(xyz)[0]
+
+    # Test on spherical input
+    xyz = random_uniform_on_sphere(n=100, coords='xyz')
+    assert not is_hemispherical(xyz)[0]
```

Comment:
Same comment as above here

Reply:
Thanks. Fixed in [4952d02](https://github.com/nipy/dipy/pull/1850/commits/4952d027bcefc3bef61c4d9bc7043a42253bbe52).